src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.hpp

```diff
@@ -272,6 +282,8 @@ class DiscoveryDataBase
 
     const fastrtps::rtps::GuidPrefix_t server_guid_prefix_;
 
+    bool server_acked_by_all_ = false;
```

Comment:
This way it's protected to read and write without locking the database
```suggestion
    std::atomic<bool> server_acked_by_all_ = false;
```